### ISSUE TITLE
having "types" as first filter

### ISSUE THREAD

**Issue by eroux** [buda-base/public-digital-library#979]
I keep doing searches for person names that are a bit obscure (ex: `ldan pa Sang rgyas`, `phug pa Blo ldan shes rab`), and I just want to know if we have a person record for them. When I search I'm getting hits in texts but I only want to see persons so I go for the "Types" filter, which unfortunately is one of the last and it often frustrates me because I believe it's a very fundamental filter. Let's put it at the top like it is in the current version of BUDA

**Comment by roopeux** [comment]
@eroux, I completely agree that it would be nice to filter persons separately from texts, but neither of your example queries return "Person" in the "Types" filter.  Should they?

I placed "Types" probably too low because with most queries it just gave 2-3 options like "Text", "Text in outline" and "Chapter", which did not seem helpful, but with some queries it does return "Person" too.

I have emailed Jann about this, just to be respectful of his earlier ideas about the filter order.  For him the "Print method" was important, for example.

**Comment by eroux** [comment]
The types filter is very fundamental. A few reasons:
- filtering by type primarily has always been a feature of the website, it the legacy form and its current BUDA form
- when doing research, users (scholars and translators) are constantly trying to answer the question "who is this person?", "what is this place?", etc. And BDRC is the largest database for that. This filter is fundamental for that kind of research, it's just that we hanven't user tested it
- in the new search we're developing, the results are excellent when there is a clear match, but when there is no match essentially display noise, sometimes many result pages of it. That's fine, but that means you have to filter the noise. To do that what I do is look at the type filter, if it doesn't show "Person", then that means I don't have to go though pages of noisy results to see if there's a person in there.

So anyway, I agree that the order of results in general is important, but let's start with making this one first again

**Comment by roopeux** [comment]
I am not against putting the Type filter on top.  @berger-n, please go ahead.

- Charles actually mentioned this briefly in his test, which proves this is an actual legacy functionality.
- My reason to be cautious was that in the given examples we would communicate by the _absence_ of an option, with the thought that this absence would be more helpful than any other filter.
- A general big red flag is to base decisions on the lead developer's own UX, especially when strong opinions arise, because very often we don't see how much our perspective differs from everyone else.

**Comment by eroux** [comment]
believe me, you made it clear that my ideas were red flags :)

**Comment by roopeux** [comment]
BDRC must be the most referenced Buddhist website, which means it has a huge value.  The website is a product of the daily stream of ideas of its creators, basically you, Jann and @berger-n, and those ideas are something to be proud of.

My job has been to guide that stream a little bit.  Whenever I myself have been on the receiving side of such guidance, it surely was irritating, so I'll offer my heartfelt thanks to the whole team for your patience and openness.

**Comment by berger-n** [comment]
thanks for all you've done @roopeux! your work has made the website much better

just moved the type filter to the top: https://library-dev.bdrc.io/?sortBy=firstScanSyncDate_desc

![image](https://github.com/user-attachments/assets/f2235ea6-33ec-47ee-b05f-f9d9c6da0e98)

---

BTW not sure what to think about these results so-called Etext type? they seem to appear in this query only
(and the corresponding data is quite empty)
https://library-dev.bdrc.io/?sortBy=firstScanSyncDate_desc&type%5B0%5D=Etext

![image](https://github.com/user-attachments/assets/6883bf4b-d472-4f2c-b0d5-fd09518f59eb)


![image](https://github.com/user-attachments/assets/0ab744ef-7f55-4b48-9b8e-7d1c24641979)

**Comment by eroux** [comment]
thanks a lot! The etext thing is a completely unrelated issue, probably more for the autocomplete repo, can you open it there?

**Comment by roopeux** [comment]
@berger-n a very big thank you to you also!  It has been a true pleasure to work with you.
